jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/AbstractMessageSink.java

```diff
@@ -18,14 +18,82 @@
 
 import org.eclipse.jetty.websocket.core.CoreSession;
 
+/**
+ * <p>Partial implementation of {@link MessageSink}.</p>
+ * <p>The application function passed to the constructor as a {@link MethodHandle}
+ * either delegates the demand for more WebSocket frames to the {@link MessageSink}
+ * implementation -- and therefore {@code autoDemand==false}, or explicitly demands
+ * for more WebSocket frames (via APIs that eventually call
+ * {@link CoreSession#demand(long)}) -- and therefore {@code autoDemand==true}.</p>
+ * <p>{@link MessageSink} implementations must handle the demand for WebSocket
+ * frames in this way:</p>
+ * <ul>
+ * <li>When {@code autoDemand==false}, the {@link MessageSink} manages the
+ * demand until the conditions to invoke the application function are met;
+ * when the {@link MessageSink} invokes the application function, then the
+ * application function is responsible to demand for more WebSocket frames.</li>
```

Comment:
```suggestion
 * <li>When {@code autoDemand==false}, the application must initiate demand
 * for a message frame and then the {@link MessageSink}, manages the
 * demand until the conditions to invoke the application function are met;
 * at which point the {@link MessageSink} invokes the application function; and then
 * the application is again responsible for initiate any demand for subsequent message frames.</li>
```

Reply:
Nope it's wrong.

What you suggest is completely irrelevant from the point of view of the `MessageSink`.
Does not matter to document here how the `MessageSink` is invoked the first time, and less so to document here how "application" code must behave.
For these javadocs needs to be documented how the `MessageSink` manages the demand after it is invoked the first time.

Reply:
why do you think the initial demand is irrelevant?   To me it is a big difference between the modes: one does the initial demand and the other doesn't.

What is wrong with my version?  I think it says much the same as yours (but better) and includes the initial demand.  

At the very least, change your version to not use "when" in two different senses.  "when the messageSink invokes..." reads like and alternative to "when autodemand==false".      Use "then" or some other language that implies a sequence of events.